src/ucs/config/parser.c

```diff
@@ -1971,21 +1967,34 @@ static void ucs_config_parser_print_env_vars(const char *prefix)
         iter = kh_get(ucs_config_env_vars, &ucs_config_parser_env_vars, var_name);
         if (iter == kh_end(&ucs_config_parser_env_vars)) {
             if (ucs_global_opts.warn_unused_env_vars) {
-                ucs_string_buffer_appendf(&unused_vars_strb, "%s", var_name);
+                ucs_string_buffer_appendf(unused_vars_strb, "%s", var_name);
                 ucs_config_parser_append_similar_vars_message(
-                        prefix, var_name, &unused_vars_strb);
-                ucs_string_buffer_appendf(&unused_vars_strb, "; ");
-                ++num_unused_vars;
+                        prefix, var_name, unused_vars_strb);
+                ucs_string_buffer_appendf(unused_vars_strb, "; ");
+                ++*num_unused_vars;
```

Comment:
1. maybe we don't have to return num_unused_vars and num_unused_vars, can use empty-string as indication
2. seems that in order to get list of env vars, we could just iterate over ucs_config_parser_env_vars hash table and save the results. so maybe no need to modify/reuse the function ucs_config_parser_get_env_vars

Reply:
1. I've used NULL as an indication if that's ok
2. We would need to append those into a single string eventually; wouldn't that duplicate most of the code?